src/System.Private.DisabledReflection/src/Internal/Reflection/RuntimeTypeInfo.cs

```diff
@@ -0,0 +1,212 @@
+
+        public override MethodInfo[] GetMethods(BindingFlags bindingAttr) => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        public override Type GetNestedType(string name, BindingFlags bindingAttr) => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        public override Type[] GetNestedTypes(BindingFlags bindingAttr) => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        public override PropertyInfo[] GetProperties(BindingFlags bindingAttr) => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        public override object InvokeMember(string name, BindingFlags invokeAttr, Binder binder, object target, object[] args, ParameterModifier[] modifiers, CultureInfo culture, string[] namedParameters)
+            => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        public override bool IsDefined(Type attributeType, bool inherit) => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        protected override TypeAttributes GetAttributeFlagsImpl() => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        protected override ConstructorInfo GetConstructorImpl(BindingFlags bindingAttr, Binder binder, CallingConventions callConvention, Type[] types, ParameterModifier[] modifiers)
+            => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        protected override MethodInfo GetMethodImpl(string name, BindingFlags bindingAttr, Binder binder, CallingConventions callConvention, Type[] types, ParameterModifier[] modifiers)
+            => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        protected override PropertyInfo GetPropertyImpl(string name, BindingFlags bindingAttr, Binder binder, Type returnType, Type[] types, ParameterModifier[] modifiers)
+             => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        protected override bool HasElementTypeImpl()
+        {
+            return RuntimeAugments.IsArrayType(_typeHandle) || RuntimeAugments.IsUnmanagedPointerType(_typeHandle) || RuntimeAugments.IsByRefType(_typeHandle);
+        }
+
+        protected override bool IsArrayImpl()
+        {
+            return RuntimeAugments.IsArrayType(_typeHandle);
+        }
+
+        protected override bool IsByRefImpl()
+        {
+            return RuntimeAugments.IsByRefType(_typeHandle);
+        }
+
+        protected override bool IsCOMObjectImpl() => throw new NotSupportedException(SR.Reflection_Disabled);
+
+        protected override bool IsPointerImpl()
+        {
+            return RuntimeAugments.IsUnmanagedPointerType(_typeHandle);
+        }
+
+        protected override bool IsPrimitiveImpl()
+        {
+            return RuntimeAugments.IsPrimitive(_typeHandle);
+        }
+
+        internal static RuntimeTypeInfo GetRuntimeTypeInfo(RuntimeTypeHandle typeHandle)
+        {
+            return RuntimeTypeTable.Table.GetOrAdd(new RuntimeTypeHandleKey(typeHandle));
```

Comment:
Do we need to reject generic type definitions here?

Reply:
I think we would only need to do a split if we decide to implement the fancier APIs such as the IsAssignableFrom you mentioned above. Generic definition EETypes cannot be passed to runtime APIs, but things like IsValueType should work as expected (template type loader relies on that).

Reply:
Make sense